sdk/core/core/inc/az_http_transport.h

```diff
@@ -85,12 +85,16 @@ AZ_NODISCARD az_result
 az_http_request_get_header(_az_http_request const* request, int32_t index, az_pair* out_header);
 
 /**
- * @brief Get parts of an HTTP request.
+ * @brief Get parts of an HTTP request. NULL in accepted to ignore getting any parts, for example,
+ * call this function like below to get only the http method and ignore getting url and body
+ *   az_http_request_get_parts(request, &method, NULL, NULL)`
+ *
+ * This function is expected to be used by transport layer only.
  *
  * @param request HTTP request to get parts from.
- * @param out_method Pointer to write HTTP method to.
- * @param out_url Pointer to write URL to.
- * @param out_body Pointer to write HTTP request body to.
+ * @param out_method Pointer to write HTTP method to. Use NULL to ignore getting this value.
```

Comment:
Annotate the parameter as nullable.
See:
https://github.com/Azure/azure-sdk-for-c/blob/da98d45fb4a1d44b940cec3e69a9358d566879c7/sdk/iot/provisioning/inc/az_iot_provisioning_client.h#L68-L70
https://azure.github.io/azure-sdk/clang_documentation.html#docstrings